concrete/src/Entity/File/Version.php

```diff
@@ -1468,21 +1482,15 @@ public function generateThumbnail(ThumbnailTypeVersion $type)
                     break;
             }
         }
-
-        $imageForThumbnail = $image;
+        if ($inplaceOperations) {
+            $thumbnailMode |= ImageInterface::THUMBNAIL_FLAG_NOCLONE;
+        }
         if ($type->isUpscalingEnabled()) {
-            $imageSize = $image->getSize();
-            if ($size->contains($imageSize) && $imageSize->getWidth() !== $size->getWidth() && $imageSize->getHeight() !== $size->getHeight()) {
-                if (($imageSize->getWidth() / $imageSize->getHeight()) >= ($size->getWidth() / $size->getHeight())) {
-                    $newImageSize = $imageSize->heighten($size->getHeight());
-                } else {
-                    $newImageSize = $imageSize->widen($size->getWidth());
-                }
-                $imageForThumbnail = $image->copy()->resize($newImageSize);
-            }
+            $thumbnailMode |= ImageInterface::THUMBNAIL_FLAG_UPSCALE;
         }
-        $thumbnail = $imageForThumbnail->thumbnail($size, $thumbnailMode);
-        unset($imageForThumbnail);
+
+        $thumbnail = $image->thumbnail($size, $thumbnailMode);
```

Comment:
Is there a reason you use `->thumbnail` rather than `->resize` as thumbnail will always make a copy of the file, unless imagine 1.1 changes this?

Reply:
Yep. Since Imagine 1.1 there's the new `ImageInterface::THUMBNAIL_FLAG_NOCLONE` flag (see https://github.com/avalanche123/Imagine/commit/2101c348f9d97074ab4dbd5dcc8fe73b7be0f5c5 and https://github.com/concrete5/concrete5/pull/7246/files#diff-c92ee3c67c5507a41aae46e1527b1d8aR1486 )